### PR TITLE
BoM Validation and downloader

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
We need a way for the customers to download the files from SAP and store them in their storage account
## Solution
This PR contains two playbooks "playbook_bom_validator" and "playbook_bom_downloader" 

**playbook_bom_validator** will validate that all URLs specified in the BoM 
**playbook_bom_downloader** will download & upload all the files based on the URLs specified in the BoM 

## Tests

These variables need to be present in the sap-parameters file
S_user:                        
S_password:               
sapbits_access_key:   

## Notes

The storage account needs to have all the containers pre-created 
In a future iteration these will be read from keyvault

### CODE REVIEW

File: deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  retries:  2
+  delay:    1
+  loop:     "{{ _bom.materials.media|flatten(levels=1) }}"
+  when:     item.url is defined
+
+#   03) Upload files to Azure
+#   Loop through BOM media files  
+- name: "BOM: Upload files"
+  command: >-
+    az storage blob upload --account-name {{ sapbits_location_base_path.rpartition('//')[2].split('.')[0]}} --account-key {{ sapbits_access_key }} --container-name {{ sapbits_location_base_path.rpartition('//')[2].split('/')[1]}}/{{ sapbits_bom_files }}/archives --name {{ item.archive }}  --file {{ download_directory }}/files/{{ item.archive }} --if-none-match "*" --no-progress
```
Comment (rtamalin) [inline deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml:82]: You can spread this long line over multiple lines (without need for end-of-line backslash separator) due to using the `>-` after the `:` on the `command:` line... E.g. put the `az storage blob upload` on the first line, and then continue with lines per parameter.

That will make it easier to maintain and read the code in the long run ;-)

Comment (KimForss) [reply deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml]: I agree - change made

File: deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml [hunk 2]
```diff
@@ -79,7 +79,14 @@
...
   command: >-
-    az storage blob upload --account-name {{ sapbits_location_base_path.rpartition('//')[2].split('.')[0]}} --account-key {{ sapbits_access_key }} --container-name {{ sapbits_location_base_path.rpartition('//')[2].split('/')[1]}}/{{ sapbits_bom_files }}/archives --name {{ item.archive }}  --file {{ download_directory }}/files/{{ item.archive }} --if-none-match "*" --no-progress
+    az storage blob upload 
+    --account-name {{ sapbits_location_base_path.rpartition('//')[2].split('.')[0]}} 
+    --account-key {{ sapbits_access_key }} 
+    --container-name {{ sapbits_location_base_path.rpartition('//')[2].split('/')[1]}}/{{ sapbits_bom_files }}/archives 
+    --name {{ item.archive }}  
+    --file {{ download_directory }}/files/{{ item.archive }} 
+    --if-none-match "*" 
+    --no-progress
```
Comment (mkdeegan) [inline deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml:89]: I would indent these for easier readability to after the "upload"

Comment (KimForss) [reply deploy/ansible/roles-sap/0.1-bom-validator/tasks/bom_validator.yaml]: done

**Pull request merged**